GVFS/GVFS.Service/GVFSService.Mac.cs

```diff
@@ -35,8 +58,19 @@ public void RunWithArgs(string[] args)
 
             try
             {
-                this.serviceThread.Start();
-                this.serviceThread.Join();
+                this.repoRegistry.Upgrade();
+
+                string pipeName = this.serviceName + ".Pipe";
+                this.tracer.RelatedInfo("Starting pipe server with name: " + pipeName);
+
+                using (NamedPipeServer pipeServer = NamedPipeServer.StartNewServer(
+                    pipeName,

```

Comment:
Related to my command in the functional tests on pipe names, can/should we use a rooted path name here so that we know exactly where the socket is?

Reply:
As discussed offline, if we don't use rooted paths here I think we might get collisions if two users have GVFS.Service processes running at the same time:

https://github.com/dotnet/corefx/blob/94e9d02ad70b2224d012ac4a66eaa1f913ae4f29/src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Unix.cs#L35

If a rooted path is not used:

> The pipe is created directly under Path.GetTempPath() with "CoreFXPipe_" prefix.

Meaning that two GVFS.Service process that both create `this.serviceName + ".Pipe"` would be trying to use the same socket.

Reply:
```
using System;
using System.IO;

namespace PrintTempPath
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine($"Temp path: {Path.GetTempPath()}");
        }
    }
}
```

Output from two separate User login sessions -
```
Temp path: /var/folders/bv/ydjjxvmn6ml_mrhyfnzjwclm0000gp/T/
Temp path: /var/folders/fk/9c4hghpx7dq33s6fx7mlgmg80000gn/T/
```

`ls -l` of the user socket path - 
```
$ ls -l /var/folders/bv/ydjjxvmn6ml_mrhyfnzjwclm0000gp/T/CoreFxPipe_GVFS.Service.Pipe 
srwxr-xr-x  1 Ameen  staff  0 Apr 15 10:19 /var/folders/bv/ydjjxvmn6ml_mrhyfnzjwclm0000gp/T/CoreFxPipe_GVFS.Service.Pipe

$ sudo ls -l /var/folders/fk/9c4hghpx7dq33s6fx7mlgmg80000gn/T/CoreFxPipe_GVFS.Service.Pipe
Password:
srwxr-xr-x  1 admin  staff  0 Apr 17 13:20 /var/folders/fk/9c4hghpx7dq33s6fx7mlgmg80000gn/T/CoreFxPipe_GVFS.Service.Pipe
```

Reply:
Also created a follow up [issue](https://github.com/Microsoft/VFSForGit/issues/1047).